packages/datatrak-web/src/features/Tasks/TaskDetails/TaskDetails.tsx

```diff
@@ -0,0 +1,235 @@
+import { useFromLocation } from '../../../utils';
+import { RepeatScheduleInput } from '../RepeatScheduleInput';
+import { DueDatePicker } from '../DueDatePicker';
+import { AssigneeInput } from '../AssigneeInput';
+import { TaskForm } from '../TaskForm';
+import { TaskMetadata } from './TaskMetadata';
+import { ROUTES } from '../../../constants';
+
+const Container = styled(Paper).attrs({
+  variant: 'outlined',
+})`
+  padding: 1.5rem;
+  display: flex;
+  flex-direction: column;
+  gap: min(2.5rem, 2%);
+  ${({ theme }) => theme.breakpoints.up('md')} {
+    flex-direction: row;
+    padding: 2.5rem;
+  }
+`;
+
+const MainColumn = styled.div`
+  display: flex;
+  flex-direction: column;
+  flex: 1;
+  margin-block: 1.2rem;
+  ${({ theme }) => theme.breakpoints.up('md')} {
+    width: 44%;
+    margin-block: 0;
+  }
+`;
+
+const SideColumn = styled.div`
+  display: flex;
+  flex-direction: column;
+  ${({ theme }) => theme.breakpoints.up('md')} {
+    width: 28%;
+  }
+`;
+
+const ItemWrapper = styled.div`
+  &:not(:last-child) {
+    margin-block-end: 1.2rem;
+  }
+`;
+
+const CommentsPlaceholder = styled(ItemWrapper)`
+  height: 10rem;
+  width: 100%;
+  border: 1px solid ${({ theme }) => theme.palette.divider};
+  border-radius: 4px;
+  ${({ theme }) => theme.breakpoints.up('md')} {
+    flex: 1;
+    height: 100%;
+  }
+`;
+
+const CommentsInput = styled(TextField).attrs({
+  multiline: true,
+  variant: 'outlined',
+  fullWidth: true,
+  rows: 4,
+})`
+  margin-block-end: 0;
+  .MuiOutlinedInput-inputMultiline {
+    padding-inline: 1rem;
+  }
+`;
+
+const ClearButton = styled(Button).attrs({
+  variant: 'text',
+})`
+  padding-inline: 0.5rem;
+`;
+
+const ButtonWrapper = styled.div`
+  display: flex;
+  justify-content: flex-end;
+  align-items: flex-end;
+  flex: 1;
+`;
+
+const Form = styled(TaskForm)`
+  .loading-screen {
+    border: 1px solid ${({ theme }) => theme.palette.divider};
+  }
+`;
+
+export const TaskDetails = () => {
+  const { taskId } = useParams();
+  const navigate = useNavigate();
+  const backLink = useFromLocation();
+  const { data: task, isLoading } = useTask(taskId);
+
+  const defaultValues = {
+    dueDate: task?.dueDate ?? null,
+    repeatSchedule: task?.repeatSchedule?.frequency ?? null,
+    assigneeId: task?.assigneeId ?? null,
+  };
+  const formContext = useForm({
+    mode: 'onChange',
+    defaultValues,
+  });
+  const {
+    control,
+    handleSubmit,
+    setValue,
+    formState: { isValid, dirtyFields },
+    reset,
+  } = formContext;
+
+  const navigateBack = () => {
+    navigate(backLink || ROUTES.TASKS);
+  };
+
+  const { mutate: editTask, isLoading: isSaving } = useEditTask(taskId, navigateBack);
+
+  useEffect(() => {
+    if (!task) return;
+    setValue('dueDate', task?.dueDate, { shouldDirty: false, shouldValidate: true });
+    setValue('repeatSchedule', task?.repeatSchedule?.frequency ?? null);
+    setValue('assigneeId', task?.assigneeId ?? null);
+  }, [JSON.stringify(task)]);
```

Comment:
It would be nice if possible to set the default values of the form rather than having a useEffect. I think that react-hook-form now provides async support for that or if that doesn't work we could just hold off on returning the form until the request is done.